### ISSUE TITLE
Support additional Javascript prolog directives

### ISSUE THREAD

**Issue by ChristianMurphy** [eslint/eslint#5655]
**What version of ESLint are you using?**
2.4.0

**What problem you want to solve?**

Currently the [strict rule](http://eslint.org/docs/rules/strict) allows for requiring or disallowing the `'use strict'` directive.
I would like to be able to do the same with other Javascript directives, such as `'use strong'` and `'use asm'`.

This would allow for validation of code like:
``` js
// global strong mode
'use strong';

console.log('example');
```

``` js
// function scoped strong mode
function example () {
    'use strong';
    console.log('example');
}
example();
```

``` js
// missing mode
function example () {
    console.log('example');
}
example();
```

``` js
// asm mode
function MyAsmModule() {
    "use asm";
    // module body
}
```

**What is your take on the correct solution to problem?**

Add a directive name option to the existing `strict` rule, that defaults to strict.

Something like:

``` yml
strict:
   - error
   - global
   - strong
```

or

``` yml
strict:
  - warn
  - scope: function
    directive: asm
```

**Additional Resources**
Related #2240
ASM JS: http://asmjs.org/spec/latest/#validatemodule-f
Strong Mode: https://github.com/v8/v8/wiki/Experiments%20with%20Strengthening%20JavaScript#strong-mode
V8/Node supported directives: https://github.com/nodejs/node/blob/v4.x/deps/v8/src/ast-value-factory.h#L283-L285

**Comment by michaelficarra** [comment]
Other directives do not belong in the `strict` rule. They are completely unrelated. If I was you, I would propose a new rule instead (which may or may not be able to subsume the functionality of `strict`). Keep in mind that rules that deal with these less-common directives have been denied before because they wouldn't affect many eslint users, so I would expect the rule to be rejected on the same grounds.

**Comment by ChristianMurphy** [comment]
Will do, Thanks @michaelficarra!
